### DISCUSSION TITLE
Cobmine text input and checkbox

### DISCUSSION THREAD

**Discussion by wardal** [react-hook-form/react-hook-form#4118] [Q&A] [answered]
Hey guys, I need your help & ideas how to better implement such form element using RHF.
My goal is to make checkbox with text input inside. When checkbox is checked, text input could be filled and this data used during formSubmit.
Small example of what I'm trying to do:
![demo](https://user-images.githubusercontent.com/15804042/107053727-d5f27b80-67d7-11eb-9108-cca1490dceba.png)

**Comment by bluebill1049** [comment, marked as answer]
build a controlled component and wrap with Controller.
